app/controllers/application_controller.rb

```diff
@@ -1,3 +1,29 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+
+  before_action :require_login
+  helper_method :logged_in?, :current_user
+
+  def set_current_user(user = nil)
```

Comment:
This method isn't a good construction - You should not save whole user in `session[:user_id]` - the problem here is also the way you name it. Also its name `set_current_user` is misguiding as you are not setting the current_user variable but the cookie. I recommend to delete whole method.